src/pybind/mgr/mgr_module.py

```diff
@@ -427,7 +428,7 @@ def get_module_option(self, key, default=None):
         :param default: the default value of the config if it is not found
```

Comment:
I'd suppress (or at least deprecate) the `default` argument here. Just for the sake of avoiding the complexity of tracking all the possible paths for an option to get a value. A default can be set when the option is defined, so is there any use case to override the 'global' default on a per call basis?

Reply:
yeah, i agree it would be nice to clean that up. that'll mean fixing all of the callers...

Reply:
We actively use this feature in the Dashboard, e.g. for the ``server_port``. We do not have separate config options for HTTP and HTTPS, so we call the ``get_localized_module_option`` method with 8443 or 8080, see https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/module.py#L134.
We should split that into separate config options anyway, but this is some more work, because the QA code requires some refactoring in this case.

Reply:
Thanks for recalling this @votdev, I missed that use case.

Reply:
There is PR https://github.com/ceph/ceph/pull/26914 that will cleanup this issue.